client/my-sites/theme/controller.jsx

```diff
@@ -9,7 +9,7 @@ import ThemeNotFoundError from './theme-not-found-error';
 const debug = debugFactory( 'calypso:themes' );
 
 export function fetchThemeDetailsData( context, next ) {
-	if ( ! context.isServerSide || context.cachedMarkup ) {
+	if ( context.cachedMarkup ) {
```

Comment:
When the context is not SSR, the fetching is disabled. This means users that are logged in and are visiting the /theme/:theme/site-name directly will get a 404 since the theme data wasn't populated.